### PR TITLE
Allow sorting by All Time views as well as Last 7 Days

### PR DESCRIPTION
**Pull request by nsharma123**
https://webarchive.jira.com/browse/WEBDEV-5004

### CODE REVIEW

File: test/sort-filter-bar/sort-filter-bar.test.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+  it('should render default view-sort selector', async () => {
+    const defaultSortSelector = sortSelectorContainer?.children
+      .item(2)
+      ?.querySelector('a');
+    expect(defaultSortSelector?.textContent).to.contain('Weekly Views');
+  });
+
+  it('should render active view-sort selectors', async () => {
+    el.selectedSort = 'alltimeview' as SortField;
```
Comment (iisa) [inline test/sort-filter-bar/sort-filter-bar.test.ts:72]: 👍

File: src/models.ts [hunk 2]
```diff
@@ -103,13 +107,14 @@ export const SortFieldToMetadataField: {
 export const MetadataFieldToSortField: {
   [key in MetadataSortField]: SortField;
 } = {
+  week: SortField.weeklyview,
+  downloads: SortField.alltimeview,
```
Comment (iisa) [inline src/models.ts:111]: good to know

File: src/sort-filter-bar/sort-filter-bar.ts [hunk 3]
```diff
@@ -235,13 +244,30 @@ export class SortFilterBar
             ? this.getSortDisplayOption(SortField.relevance)
             : nothing}
         </li>
-        <li>${this.getSortDisplayOption(SortField.views)}</li>
+        <li>
+          ${this.getSortDisplayOption(SortField.weeklyview, {
+            clickEvent: () => {
+              if (!this.viewOptionSelected)
+                this.setSelectedSort(SortField.weeklyview);
```
Comment (iisa) [inline src/sort-filter-bar/sort-filter-bar.ts:251]: Q: is this the default view?

File: src/sort-filter-bar/sort-filter-bar.ts [hunk 4]
```diff
@@ -414,6 +441,26 @@ export class SortFilterBar
...
+      ></div>
+      <div id="view-sort-selector">
+        <ul>
+          <li>${this.getDateSortButton(SortField.alltimeview)}</li>
+          <li>${this.getDateSortButton(SortField.weeklyview)}</li>
+        </ul>
+      </div>
+    `;
+  }
+
```
Comment (iisa) [inline src/sort-filter-bar/sort-filter-bar.ts:463]: codecov warning here, how can we test this indirectly?

**Review by iisa — APPROVED** [review]
looks good. just a couple of questions, comments

thanks for bulking up the unit tests with this pass 💪

File: src/sort-filter-bar/sort-filter-bar.ts (hunk 3 above)
Comment (nsharma123) [reply src/sort-filter-bar/sort-filter-bar.ts:251]: Yes

**Pull request merged**